core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -344,6 +344,9 @@ public int from() {
      * The number of search hits to return. Defaults to <tt>10</tt>.
      */
     public SearchSourceBuilder size(int size) {
+        if (size < 0) {
+            throw new IllegalArgumentException("[size] parameter cannot be negative");
+        }
```

Comment:
Maybe we can set the size to the default of 10, that's what `SearchService#createContext` defaults to later on anyway if it finds the current `-1` value. I just find it easier to reason about if the value is set to the default in this case.